rust/processor/src/processors/econia_processor.rs

```diff
@@ -618,14 +651,76 @@ impl ProcessorTrait for EconiaTransactionProcessor {
                     )?);
                 }
             }
+            // Index transaction write set.
+            let info = &txn.info.as_ref().expect("No transaction info");
+            for change in &info.changes {
+                match change.change.as_ref().expect("No transaction changes") {
+                    Change::WriteResource(resource) => {
+                        if resource.r#type.as_ref().expect("No resource type")
+                            == &market_accounts_type
+                        {
+                            let data: serde_json::Value = serde_json::from_str(&resource.data)
+                                .expect("Failed to parse MarketAccounts");
+                            market_account_handles.push(MarketAccountHandle {
+                                user: resource.address.clone(),
+                                handle: data["map"]["handle"].as_str().unwrap().to_string(),
+                            })
+                        }
+                    },
+                    Change::WriteTableItem(write) => {
+                        let table_data = write.data.as_ref().expect("No WriteTableItem data");
+                        if table_data.value_type
+                            != format!("{}::user::MarketAccount", econia_address)
+                        {
+                            continue;
+                        }
+                        let table_key: serde_json::Value = serde_json::from_str(&table_data.key)
+                            .expect("Failed to parse market account ID");
+                        let market_account_id =
+                            u128::from_str(table_key.as_str().unwrap()).unwrap();
+                        let data: serde_json::Value = serde_json::from_str(&table_data.value)
+                            .expect("Failed to parse MarketAccount");
```

Comment:
Let's avoid `unwrap` here as well. See `opt_value_to_*` for examples on how to do that.

Reply:
Unwraps removed per 306a3cb